### PR TITLE
Implemented Required RadioGroups

### PR DESCRIPTION
**Pull request by gulnazsayed**
### Platforms Impacted
- [ ] iOS
- [ ] macOS
- [x] win32 (Office)
- [ ] windows
- [ ] android

### Description of changes

Implemented required prop and styling for required RadioGroups. Added a section for required RadioGroups to the tester page.

### Verification

(how the change was tested, including both manual and automated tests)

| Before                                       | After                                      |
|----------------------------------------------|--------------------------------------------|
| image | 
image |

### Pull request checklist

This PR has considered (when applicable):
- [ ] Automated Tests
- [ ] Documentation and examples
- [ ] Keyboard Accessibility
- [ ] Voiceover
- [ ] Internationalization and Right-to-left Layouts

### CODE REVIEW

**Comment by rurikoaraki** [comment]
Out of curiosity how does the PR have 71 commits? Are some of these from things you've already merged into main?

**Comment by samuelfreiberg** [comment]
Just wondering - do we have a tooltip or some type of popup that happens when you hover required RadioGroups? Thinking about it from a user's perspective, a `*` might not always be so obvious that it means required. It might help to have a tooltip that communicates this.

Or, I guess if a Tooltip is an optional prop on RadioGroup, it's up to the consumer to input that.

File: change/@fluentui-react-native-experimental-radio-group-346f9bef-d345-48b5-b108-de0a1146e8fd.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (samuelfreiberg) [inline change/@fluentui-react-native-experimental-radio-group-346f9bef-d345-48b5-b108-de0a1146e8fd.json:1]: Is this relevant to this PR? It looks like an old changelong? This comment doesn't seem to deal with this PRs new feature

File: change/@fluentui-react-native-tester-714274b6-760e-4c75-9e5f-f9bb456ff3d6.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "minor",
```
Comment (samuelfreiberg) [inline change/@fluentui-react-native-tester-714274b6-760e-4c75-9e5f-f9bb456ff3d6.json:2]: Same with this. This Changelog doesn't seem to be relevant to this PR's change

**Review by samuelfreiberg — CHANGES_REQUESTED** [review]

File: change/@fluentui-react-native-experimental-radio-group-346f9bef-d345-48b5-b108-de0a1146e8fd.json (hunk 1 above)
Comment (gulnazsayed) [reply change/@fluentui-react-native-experimental-radio-group-346f9bef-d345-48b5-b108-de0a1146e8fd.json]: Thanks for catching this. I think it carried over from an older local branch that I had forked from, which is also why there are so many old commits in the history. Will fix this for future PRs by forking directly from the main branch.

**Comment by gulnazsayed** [comment]
@samuelfreiberg Are there different types of tooltips? Is there one you think we could use specifically for required? In my experience, I don't think I've seen a tooltip that implies 'required' before. Visually, web doesn't have any other indication besides the '*' either.

**Comment by rurikoaraki** [comment]
> @rurikoaraki I created this branch from a local branch instead of main since the Radio and RadioGroup framework implementations weren't merged into main yet at the time. I don't think there's an easy way for me to remove the old commits without creating a new branch that's based from main. I'll make sure to create future branches and PRs with the base as main to avoid this issue in the future! Thanks for the feedback

What I generally do is I'll rebase a branch before pushing it once the changes I'm building on top of are in main. Merging main into the branch is fine too. Really what I'd like to see is the changes you made specifically to make this feature work, and to not see other changes mixed in with that (which I thought I saw when I first looked at this PR). Minimal commits is nice for tools like codeflow, but not a requirement,

File: packages/experimental/RadioGroup/src/RadioGroup/RadioGroup.tsx [hunk 3]
```diff
@@ -28,7 +30,16 @@ export const RadioGroup = compose<RadioGroupType>({
         return null;
       }
 
-      const { label, defaultTabbableElement, isCircularNavigation, ...mergedProps } = mergeProps(radioGroup.props, final);
+      const { label, required, defaultTabbableElement, isCircularNavigation, ...mergedProps } = mergeProps(radioGroup.props, final);
+
+      const labelComponent = (
+        <React.Fragment>
```
Comment (rurikoaraki) [inline packages/experimental/RadioGroup/src/RadioGroup/RadioGroup.tsx:36]: I don't think this needs to be wrapped in a Fragment

File: packages/experimental/RadioGroup/src/RadioGroup/useRadioGroup.ts [hunk 4]
```diff
@@ -20,13 +20,15 @@ export const useRadioGroup = (props: RadioGroupProps): RadioGroupInfo => {
...
     updateSelectedButtonRef: onSelectButtonRef,
   };
 
   return {
     props: {
       ...props,
+      required,
       accessible: true,
       accessibilityRole: 'radiogroup',
       accessibilityLabel: accessibilityLabel ?? label,
```
Comment (rurikoaraki) [inline packages/experimental/RadioGroup/src/RadioGroup/useRadioGroup.ts:34]: I think there needs to be a required accessibilityState set for screen readers to correctly announce the required state of a radio group. You can look at Checkbox as an example.

File: packages/experimental/RadioGroup/src/RadioGroup/radioGroupContext.ts [hunk 5]
```diff
@@ -8,6 +8,7 @@ export interface RadioGroupContextValue extends RadioGroupState {}
 
 export const RadioGroupContext = React.createContext<RadioGroupContextValue>({
   value: null,
+  required: false,
```
Comment (rurikoaraki) [inline packages/experimental/RadioGroup/src/RadioGroup/radioGroupContext.ts:11]: Why is this neeed in the context?

File: packages/experimental/RadioGroup/src/RadioGroup/RadioGroup.tsx [hunk 6]
```diff
@@ -28,7 +30,16 @@ export const RadioGroup = compose<RadioGroupType>({
...
+          <Slots.label>
+            <Slots.labelText key="label">{label}</Slots.labelText>
+            {!!required && <Slots.required>{typeof required === 'string' ? required : '*'}</Slots.required>}
```
Comment (rurikoaraki) [inline packages/experimental/RadioGroup/src/RadioGroup/RadioGroup.tsx:39]: The required prop is currently only type boolean, is it meant to support strings, too? If it's only supporting boolean you don't need the "typeof" conditional

File: packages/experimental/RadioGroup/src/RadioGroup/RadioGroup.types.ts [hunk 7]
```diff
@@ -32,7 +33,22 @@ export interface RadioGroupState extends RadioGroupProps {
...
+
+  /**
+   * Amount of padding between the end of the label and the start of the required text
+   */
+  requiredPadding?: ViewStyle['padding'];
+
+  /**
+   * States that can be applied to a RadioGroup
+   */
+  required?: RadioGroupTokens;
```
Comment (rurikoaraki) [inline packages/experimental/RadioGroup/src/RadioGroup/RadioGroup.types.ts:50]: Why does required need to have its own state?

File: packages/experimental/RadioGroup/src/RadioGroup/RadioGroup.tsx (hunk 3 above)
Comment (gulnazsayed) [reply packages/experimental/RadioGroup/src/RadioGroup/RadioGroup.tsx:36]: Removed!

File: packages/experimental/RadioGroup/src/RadioGroup/useRadioGroup.ts (hunk 4 above)
Comment (gulnazsayed) [reply packages/experimental/RadioGroup/src/RadioGroup/useRadioGroup.ts:34]: Added!

File: packages/experimental/RadioGroup/src/RadioGroup/radioGroupContext.ts (hunk 5 above)
Comment (gulnazsayed) [reply packages/experimental/RadioGroup/src/RadioGroup/radioGroupContext.ts:11]: In the web implementation, Radios listen and inherit the prop from RadioGroup. Stylistically and functionally, there's no change to Radio when required is true, but I'm guessing we still want to use it to report accessibilityState of Radios too.

File: packages/experimental/RadioGroup/src/RadioGroup/RadioGroup.tsx (hunk 6 above)
Comment (gulnazsayed) [reply packages/experimental/RadioGroup/src/RadioGroup/RadioGroup.tsx:39]: Hm, I matched this to the web implementation but, looking at it again, web also only supports boolean for required so I'm not sure how this works. I was assuming they had this to allow the user to pass in a different indication in replacement of the '*'. Removing the conditional for now because I don't think there's a current use case for it.

File: packages/experimental/RadioGroup/src/RadioGroup/RadioGroup.types.ts (hunk 7 above)
Comment (gulnazsayed) [reply packages/experimental/RadioGroup/src/RadioGroup/RadioGroup.types.ts]: Thanks for catching this! I was using this to set the required colors before I created a separate slot for it.

File: packages/experimental/RadioGroup/src/RadioGroup/radioGroupContext.ts (hunk 5 above)
Comment (gulnazsayed) [reply packages/experimental/RadioGroup/src/RadioGroup/radioGroupContext.ts]: I agree, I'll remove it from Radio and the context.

File: packages/experimental/RadioGroup/src/Radio/useRadio.ts [hunk 8]
```diff
@@ -2,6 +2,8 @@ import { RadioProps, RadioState } from './Radio.types';
 import * as React from 'react';
 import { useRadioGroupContext } from '../RadioGroup/radioGroupContext';
 import { useAsPressable, useOnPressWithFocus, useViewCommandFocus } from '@fluentui-react-native/interactive-hooks';
+import { memoize } from '@fluentui-react-native/framework';
```
Comment (rurikoaraki) [inline packages/experimental/RadioGroup/src/Radio/useRadio.ts:5]: You shouldn't need to change this file for required I think, some of these changes are going to conflict with your other PR.

Comment (gulnazsayed) [reply packages/experimental/RadioGroup/src/Radio/useRadio.ts]: Merged and fixed!

File: packages/experimental/RadioGroup/src/RadioGroup/RadioGroup.styling.ts [hunk 9]
```diff
@@ -13,13 +12,32 @@ export const stylingSettings: UseStylingOptions<RadioGroupProps, RadioGroupSlotP
...
+        alignItems: 'center',
+        flexDirection: 'row',
+        alignSelf: 'flex-start',
+      },
+    })),
+    labelText: buildProps(
+      (tokens: RadioGroupTokens) => ({
+        variant: tokens.variant,
+        style: {
+          color: tokens.color,
```
[the remark this answers is not in the document]
Comment (gulnazsayed) [reply packages/experimental/RadioGroup/src/RadioGroup/RadioGroup.styling.ts]: Merged from other PR

**Review by rurikoaraki — APPROVED** [review]

**Review by samuelfreiberg — APPROVED** [review]
